lib/iris/experimental/um.py

```diff
@@ -685,19 +774,43 @@ def normalise(values, kind):
             output_file.seek((header.data_start - 1) * self._word_size)
             dataset_type = self.fixed_length_header.dataset_type
             sector_size = self._WORDS_PER_SECTOR * self._word_size
+
             for field in self.fields:
-                data = field.get_data()
-                if data is not None:
+                if isinstance(field, (Field2, Field3)):
+                    # Output 'recognised' lookup types (not blank entries).
                     field.lbegin = output_file.tell() / self._word_size
-                    # Round the data length up to the nearest whole
-                    # number of "sectors".
-                    size = data.size
-                    size -= size % -self._WORDS_PER_SECTOR
-                    field.lbnrec = size
-                    kind = {1: 'f', 2: 'i', 3: 'i'}.get(field.lbuser1,
-                                                        data.dtype.kind)
-                    data = normalise(data, kind)
-                    output_file.write(data)
+                    required_lbpack = field.lbpack
+                    field._check_valid_data_encoding(required_lbpack)
+                    if field._raw_data_is_useable_with_lbpack(required_lbpack):
```

Comment:
I don't fully follow the two methods here (I'm not sure they are describing their function fully). Naively, I imagined just doing:

```
if field._raw_data_must_be_unpacked_to_save(target_lbpack, target_bpacc)
    # write_bytes...
else:
    field._assert_data_can_be_written(target_lbpack)
    # do the work...
```